crystal_toolkit/core/mpcomponent.py

```diff
@@ -603,41 +602,120 @@ def get_choice_input(
-            ],
-            data=[{"key": k, "value": v} for k, v in default.items()],
-            editable=True,
-            persistence=False,
-        )
+        default = default or state.get(kwarg_label, {})
+
+        dict_size = len(default) or dict_size or 0
+
+        style = {
+            "textAlign": "center",
+            "width": "20rem",
+            "marginRight": "0.2rem",
+            "marginBottom": "0.2rem",
+            "height": "36px",
+        }
+        if "style" in kwargs:
+            style.update(kwargs.pop("style"))
+
+        def pair_element(idx, key=None, value=None):
+            # TODO: maybe move element out of the name
+            kid = self.id(kwarg_label, is_kwarg=True, idx=("k", idx), hint="dict")
+            key_in = dcc.Input(
+                id=kid,
+                debounce=True,
+                className="input",
+                style=style,
+                value=value,
+                persistence=True,
+                **kwargs,
+            )
+            vid = self.id(kwarg_label, is_kwarg=True, idx=("v", idx), hint="dict")
+            value_in = dcc.Input(
+                id=vid,
+                debounce=True,
+                className="input",
+                style=style,
+                value=value,
+                persistence=True,
+                **kwargs,
+            )
+
+            return [key_in, value_in]
+
+        # arrange the input boxes in two columns
+        dict_div_contents = [html.Div(H6(f"{key_name}: {value_name}"))]
+        # dict_div_contents = []
+        for n_idx, k in zip_longest(range(dict_size), default.keys()):
+            dict_div_contents.append(html.Div(pair_element(n_idx, k, default.get(k))))
+
+        dict_input = html.Div(dict_div_contents)
 
         return add_label_help(dict_input, label, help_str)
 
+    # def get_dict_input(
+    #     self,
+    #     kwarg_label: str,
+    #     default: Any | None = None,
+    #     state: dict | None = None,
+    #     label: str | None = None,
+    #     help_str: str = None,
+    #     key_name: str = "key",
+    #     value_name: str = "value",
+    # ):
+    #     """
+    #
+    #     :param kwarg_label:
+    #     :param default:
+    #     :param state:
+    #     :param label:
+    #     :param help_str:
+    #     :param key_name:
+    #     :param value_name:
+    #     :return:
+    #     """
+    #
+    #     state = state or {}
+    #     default = default or state.get(kwarg_label) or {}
+    #
+    #     dict_input = dt.DataTable(
+    #         id=self.id(kwarg_label, is_kwarg=True, hint="dict"),
+    #         columns=[
+    #             {"id": "key", "name": key_name},
+    #             {"id": "value", "name": value_name},
+    #         ],
+    #         data=[{"key": k, "value": v} for k, v in default.items()],
+    #         editable=True,
+    #         persistence=False,
+    #     )
+    #
+    #     return add_label_help(dict_input, label, help_str)
+
```

Comment:
I think we can remove the old implementation?

Reply:
yes, I removed it.